### PR TITLE
add mock for BGS svc

### PR DESCRIPTION
**Pull request by greene858**
This is a mock of the BGS API based on the bgs wsdl using CastleMock
Associated tickets or Slack threads:
- #1354 

This replaces PR #1420

### CODE REVIEW

File: mocks/mock-bgs-api/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+server_port=20500
```
Comment (yoomlam) [inline mocks/mock-bgs-api/gradle.properties:1]: Comment it out to clearly denote that it isn't being used
```suggestion
# server_port=20500
```

File: mocks/mock-bgs-api/src/docker/Dockerfile
Comment (yoomlam) [inline mocks/mock-bgs-api/src/docker/Dockerfile:1]: Move Dockerfile to `mocks/mock-bgs-api`. See https://github.com/department-of-veterans-affairs/abd-vro/blob/8a29a810c1d1b13217b7c9c2a764523b999fa3f0/mocks/mock-bie-kafka/build.gradle

File: mocks/mock-bgs-api/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+healthcheck_port=20501
```
Comment (yoomlam) [inline mocks/mock-bgs-api/gradle.properties:2]: This isn't being used either
```suggestion
# healthcheck_port=20501
```

File: mocks/mock-bgs-api/src/main/resources/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+#turn on bash job control
+set -m
```
Comment (yoomlam) [inline mocks/mock-bgs-api/src/main/resources/entrypoint.sh:4]: Describe why this is needed.

File: mocks/mock-bgs-api/src/main/resources/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+#Start the helper process
+/init-after-tomcat-starts.sh &
```
Comment (yoomlam) [inline mocks/mock-bgs-api/src/main/resources/entrypoint.sh:7]: Describe why this process is run in the background.

File: mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+until [ "`curl --silent --show-error --connect-timeout 1 -I http://localhost:8080/castlemock/ | grep 'HTTP/1.1 200'`" != "" ];
+do
+  echo --- waiting on castlemock for 3 seconds
```
Comment (yoomlam) [inline mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh:5]:
```suggestion
  echo "--- waiting on castlemock for 3 seconds"
```

File: mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+until [ "`curl --silent --show-error --connect-timeout 1 -I http://localhost:8080/castlemock/ | grep 'HTTP/1.1 200'`" != "" ];
```
Comment (yoomlam) [inline mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh:3]: To get the status code, try this more reliable and direct approach instead -- see https://superuser.com/questions/272265/getting-curl-to-output-http-status-code/442395#442395 and https://github.com/department-of-veterans-affairs/abd-vro/blob/89aa4b9ad4eae5eff02e2b1b075345f155edcd5d/.github/workflows/xample-integration-test.yml#L49-L52:
```suggestion
until [ "$(curl --silent -o /dev/null -w "%{http_code}" --connect-timeout 1 -I http://localhost:8080/castlemock/)" == "200"
```

File: mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+do
+  echo --- waiting on castlemock for 3 seconds
+  sleep 3
+done
+
+echo castlemock is ready!
```
Comment (yoomlam) [inline mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh:9]:
```suggestion
echo "castlemock is ready!"
```

File: mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh
Comment (yoomlam) [inline mocks/mock-bgs-api/src/main/resources/init-after-tomcat-starts.sh:1]: I interpreted the filename as "run this script after Tomcat starts" when really it's the script that is waiting until Tomcat starts How about renaming to something like `init-castlemock.sh`. The waiting aspect is an implementation detail that doesn't need to be exposed in the filename, and the script's goal is on initializing CastleMock -- Tomcat just happens to be the application server, another implementation detail.

File: mocks/mock-bgs-api/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  </vaws:VaServiceHeaders>
+</wsse:Security>
+</env:Header><env:Body><tns:createNote><note><bnftClmNoteTc>CLMDVLNOTE</bnftClmNoteTc><clmId>600391838
+</clmId><createDt>2023-03-22T19:39:05Z</createDt><noteOutTn>Claim Development Note</noteOutTn><ptcpntId>600033542</ptcpntId><txt>Note 3 from VRO</txt><userId>601174530</userId></note></tns:createNote></env:Body></env:Envelope>  
+```
+
+and curl:
+```agsl
+curl -v -H 'SOAPAction: "createNote"' -H 'Content-Type: text/xml;charset=UTF-8' --data-binary @request.xml http://localhost:20500/castlemock/mock/soap/project/x5KVzK/DevelopmentNotesService
+```
```
Comment (yoomlam) [inline mocks/mock-bgs-api/README.md:45]: I suggest adding a note that this mock simulates the happy path. To simulate an error, we'd need to tell it to return an error when a particular request is submitted, similar to unit test mocks.

**Comment by yoomlam** [comment]
@greene858 Please create a ticket to build an integration test that shows our BGS microservice using this mock.

File: mocks/docker-compose.yml [hunk 9]
```diff
@@ -102,6 +102,17 @@ services:
     networks:
       - vro_intranet
 
+  mock-bgs-api:
+    profiles: ["all","bgs"]
+    image: va/vro_mocks-mock-bgs-api:latest
+    <<: [*common-sde-security, *common-security-opt]
+    ports:
+      - "20500:8080"
```
[the remark this answers is not in the document]
Comment (greene858) [reply mocks/docker-compose.yml:110]: Done!

**Comment by msnwatson** [comment]
@greene858 is there still a need for this PR?

**Comment by dfitchett** [comment]
Closing since yoom is gone. Will replace with new PR.

**Pull request closed without merging**